frontend/src/metabase/search/components/InfoText/use-info-text.tsx

```diff
@@ -0,0 +1,160 @@
+import { useMemo } from "react";
+import { t } from "ttag";
+import {
+  PLUGIN_COLLECTION_COMPONENTS,
+  PLUGIN_COLLECTIONS,
+} from "metabase/plugins";
+import type { WrappedResult } from "metabase/search/types";
+import type { Collection } from "metabase-types/api";
+import { useDatabaseQuery, useTableQuery } from "metabase/common/hooks";
+import {
+  browseDatabase,
+  browseSchema,
+  collection as collectionUrl,
+  tableRowsQuery,
+} from "metabase/lib/urls";
+
+import { Box } from "metabase/ui";
+import type TableType from "metabase-lib/metadata/Table";
+
+const { CollectionAuthorityLevelIcon } = PLUGIN_COLLECTION_COMPONENTS;
+
+export type InfoTextData = {
+  link?: string | null;
+  icon?: JSX.Element | null;
+  label: string | null;
+};
+
+export const useInfoText = (result: WrappedResult): InfoTextData[] => {
+  let infoTextHook;
+
+  switch (result.model) {
+    case "table":
+      infoTextHook = useTablePath;
+      break;
+    case "segment":
+    case "metric":
+      infoTextHook = useTableLink;
+      break;
+    case "collection":
+      infoTextHook = useCollectionInfoText;
+      break;
+    case "database":
+      infoTextHook = useDatabaseInfoText;
+      break;
+    case "action":
+      infoTextHook = useActionInfoText;
+      break;
+    case "card":
+    case "dataset":
+    case "indexed-entity":
+    default:
+      infoTextHook = useCollectionResultLink;
+      break;
+  }
+
+  return infoTextHook(result);
+};
+const useActionInfoText = (result: WrappedResult): InfoTextData[] => {
```

Comment:
Most of these functions are not really hooks, they're simple functions. Their names should not be prefixed with `use`.